test/e2e/log_size_monitoring.go

```diff
@@ -0,0 +1,219 @@
+const (
+	// Minimal period between polling log sizes from components
+	pollingPeriod            = 5 * time.Second
+	workersNo                = 5
+	kubeletLogsPath          = "/var/log/kubelet.log"
+	kubeProxyLogsPath        = "/var/log/kube-proxy.log"
+	kubeAddonsLogsPath       = "/var/log/kube-addons.log"
+	kubeMasterAddonsLogsPath = "/var/log/kube-master-addons.log"
+	apiServerLogsPath        = "/var/log/kube-apiserver.log"
+	controllersLogsPath      = "/var/log/kube-controller-manager.log"
+	schedulerLogsPath        = "/var/log/kube-scheduler.log"
+)
+
+var (
+	nodeLogsToCheck   = []string{kubeletLogsPath, kubeProxyLogsPath}
+	masterLogsToCheck = []string{kubeletLogsPath, kubeAddonsLogsPath, kubeMasterAddonsLogsPath,
+		apiServerLogsPath, controllersLogsPath, schedulerLogsPath}
+)
+
+// LogsSizeVerifier gathers data about log files sizes from master and node machines.
+// It oversees a <workersNo> workers which do the gathering.
+type LogsSizeVerifier struct {
+	client       *client.Client
+	closeChannel chan bool
+	data         map[string]map[string]*LogsSizeData
+	lock         *sync.Mutex
+	masterIP     string
+	nodes        []string
+	wg           *sync.WaitGroup
+	workChannel  chan WorkItem
+	workers      []*LogSizeGatherer // Need at least two workers, as worker 0 is dedicated to master
+}
+
+// LogSizeGatherer is a worker which grabs a WorkItem from the channel and does assigned work
+type LogSizeGatherer struct {
+	closeChannel chan bool
+	data         *map[string]map[string]*LogsSizeData
+	lock         *sync.Mutex
+	wg           *sync.WaitGroup
+	workChannel  chan WorkItem
+}
+
+// TimestampedSize contains a size together with a time of measurement.
+type TimestampedSize struct {
+	timestamp time.Time
+	size      int
+}
+
+// LogsSizeData keeps whole timesries of TimestampedSize.
+type LogsSizeData struct {
+	sizeTimeseries []TimestampedSize
+}
+
+// WorkItem is a command for a worker that containes an IP of machine from which we want to
+// gather data and paths to all files we're interested in.
+type WorkItem struct {
+	ip    string
+	paths []string
+}
+
+// NewLogsVerifier creates a new LogsSizeVerifier which will stop when closeChannel is closed
+func NewLogsVerifier(c *client.Client, closeChannel chan bool) *LogsSizeVerifier {
+	nodes, err := NodeSSHHosts(c)
+	expectNoError(err)
+
+	data := make(map[string]map[string]*LogsSizeData)
+	lock := &sync.Mutex{}
+	wg := &sync.WaitGroup{}
+	wg.Add(workersNo)
+	workChannel := make(chan WorkItem, len(nodes)+1)
+	workers := make([]*LogSizeGatherer, workersNo)
+
+	verifier := &LogsSizeVerifier{
+		client:       c,
+		closeChannel: closeChannel,
+		data:         data,
+		lock:         lock,
+		masterIP:     getMasterHost() + ":22",
+		nodes:        nodes,
+		wg:           wg,
+		workChannel:  workChannel,
+		workers:      workers,
+	}
+	for i := 0; i < workersNo; i++ {
+		workers[i] = &LogSizeGatherer{
+			closeChannel: closeChannel,
+			data:         &verifier.data,
+			lock:         lock,
```

Comment:
Or even better one more time:
- add this thread-safe structure
- and pass the function ThisStructure.AppendData to the worker.